code/common/grafana.q

```diff
@@ -0,0 +1,151 @@
+\d .grafana
+
+// user defined column name of time column
+timeCol:@[value;`.grafana.timeCol;`time];
+// user defined column name of sym column
+sym:@[value;`.grafana.sym;`sym];
+// user defined date range to find syms from
+timeBackdate:@[value;`.grafana.timeBackdate;2D];
+// user defined number of ticks to return
+ticks:@[value;`.grafana.ticks;1000];
+// user defined query argument deliminator
+del:@[value;`.grafana.del;"."];
+
+// json types of kdb datatypes
+types:.Q.t!`array`boolean,(3#`null),(5#`number),11#`string;
+// milliseconds between 1970 and 2000
+epoch:946684800000;
+
+// wrapper if user has custom .z.pp
+.z.pp:{[f;x]$[(`$"X-Grafana-Org-Id")in key last x;zpp;f]x}[@[value;`.z.pp;{{[x]}}]];
+
+// return alive response for GET requests
+.z.ph:{[f;x]"HTTP/1.1 200 OK\r\nConnection: close\r\n\r\n";f x}[@[value;`.z.ph;{{[x]}}]];
+
+// retrieve and convert Grafana HTTP POST request then process as either timeseries or table
+zpp:{
+  // get API url from request
+  r:" " vs first x;
+  // convert grafana mesage to q
+  rqt:.j.k r 1;
+  $["query"~r 0;query[rqt];"search"~r 0;search rqt;`$"Annotation url nyi"]
+ };
+
+query:{[rqt]
+  // retrieve final query and append to table to log
+  rqtype:raze rqt[`targets]`type;
+  :.h.hy[`json]$[rqtype~"timeserie";tsfunc rqt;tbfunc rqt];
+ };
+
+finddistinctsyms:{?[x;enlist(>;timeCol;(-;.z.p;timeBackdate));1b;{x!x}enlist sym]sym};
+
+search:{[rqt]
+  // build drop down case options from tables in port
+  tabs:tables[];
+  symtabs:tabs where sym in'cols each tabs;
+  timetabs:tabs where timeCol in'cols each tabs;
+  rsp:string tabs;
+  if[count timetabs;
+    rsp,:s1:("t",del),/:string timetabs;
+    rsp,:s2:("g",del),/:string timetabs; 
+    rsp,:raze(s2,'del),/:'c1:string {(cols x) where`number=types (0!meta x)`t}each timetabs;
+    rsp,:raze((("o",del),/:string timetabs),'del),/:'c1;
+    if[count symtabs;
+      rsp,:raze(s1,'del),/:'c2:string each finddistinctsyms'[timetabs];
+      rsp,:raze((("o",del),/:string timetabs),'del),/:'{x[0] cross del,'string finddistinctsyms x 1}each (enlist each c1),'timetabs;
+     ];
+   ];
+  :.h.hy[`json].j.j rsp;
+ };
+
+diskvals:{c:count[x]-ticks+til ticks;get'[.Q.ind[x;c]]};
```

Comment:
I think this should be 
c:(count[x]-ticks)+til ticks

get' syntax is a bit confusing, would be more usual to do flip value flip e.g. 

q)\ts get each .Q.ind[trade;100+til 100000]
54 30712816
q)\ts flip value flip .Q.ind[trade;100+til 100000]
35 30712608
q)(flip value flip .Q.ind[trade;100+til 100000]) ~ get each .Q.ind[trade;100+til 100000]
1b